File: src/demo-ts/calculator.ts
export class CalculatorDivisionByZeroError extends Error {}

export class Calculator {
  constructor(public readonly stack: number) {}

  add(value: number): Calculator {
    return new Calculator(this.stack + value);
  }

  sub(value: number): Calculator {
    return new Calculator(this.stack - value);
  }

  mul(value: number): Calculator {
    return new Calculator(this.stack * value);
  }

  div(value: number): Calculator {
    if (value === 0) {
      throw new CalculatorDivisionByZeroError();
    }

    return new Calculator(this.stack / value);
  }
}

export default Calculator;

File: src/demo/calculator.py
from math import isclose


class CalculatorDivisionByZeroError(Exception):
    ...


class Calculator:
    def __init__(self, stack: float = 0) -> None:
        self.stack = stack

    def add(self, number: float) -> "Calculator":
        self.stack += number

        return self

    def sub(self, number: float) -> "Calculator":
        self.stack -= number

        return self

    def mul(self, number: float) -> "Calculator":
        self.stack *= number

        return self

    def div(self, number: float) -> "Calculator":
        if number == 0:
            raise CalculatorDivisionByZeroError

        self.stack /= number

        return self

    def __eq__(self, number: float) -> bool:
        return isclose(self.stack, number)
